File: test-suite/src/test/java/io/micronaut/docs/hibernate/proxies/Pet.java
package io.micronaut.docs.hibernate.proxies;

// tag::imports[]
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToOne;
// end::imports[]

// tag::clazz[]
@Entity
public class Pet {

    @Id
    @GeneratedValue
    private Long id;

    private String name;

    @ManyToOne(fetch = FetchType.LAZY)
    private Owner owner;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Owner getOwner() {
        return owner;
    }

    public void setOwner(Owner owner) {
        this.owner = owner;
    }
}
// end::clazz[]

File: test-suite-kotlin/src/test/kotlin/io/micronaut/docs/hibernate/proxies/Pet.kt
package io.micronaut.docs.hibernate.proxies

// tag::imports[]
import jakarta.persistence.Entity
import jakarta.persistence.FetchType
import jakarta.persistence.GeneratedValue
import jakarta.persistence.Id
import jakarta.persistence.ManyToOne
// end::imports[]

// tag::clazz[]
@Entity
open class Pet {

    @Id
    @GeneratedValue
    open var id: Long? = null

    open var name: String? = null

    @ManyToOne(fetch = FetchType.LAZY)
    open var owner: Owner? = null
}
// end::clazz[]
